### PR TITLE
Pulsar docs

### PR DESCRIPTION
**Pull request by antonio2368**

### CODE REVIEW

**Review by g-despot — APPROVED** [review]

File: docs/reference-guide/overview.md [hunk 1]
```diff
@@ -9,68 +9,86 @@ The **Memgraph** reference guide contains the details of implemented features.
...
-Memgraph supports replication and allows you to create a cluster of nodes running Memgraph instances.
+Memgraph comes with a couple of queries that allow you to safely make a backup
+of the files containing its data. Memgraph supports replication and allows you
+to create a cluster of nodes running Memgraph instances.
 
 ## [Streams](/reference-guide/streams/overview.md)
 
-Memgraph can connect to existing Kafka streams. The streams are using user-defined transformations to produce Cypher queries based on the received messages.
-To learn more about streams and transformations take a look at the following pages:
+Memgraph can connect to existing Kafka streams. The streams are using
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/overview.md:64]:
```suggestion
Memgraph can connect to existing Kafka and Pulsar streams. The streams are using
```

File: docs/reference-guide/streams/overview.md [hunk 2]
```diff
@@ -40,18 +54,42 @@ transform procedure|Name of the transformation file followed by a function name|
...
+  [BATCH_INTERVAL <batch interval length>]
+  [BATCH_SIZE <batch size>]
+  [SERVICE_URL <service url>];
+
+```
+
+option|description|type|example|default
+:-:|:-:|:-:|:-:|:-:
+stream name|Name of the stream in Memgraph|plain text|my_stream|/
+topic|Name of the topic in Kafka|plain text|my_topic|/
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/streams/overview.md:75]:
```suggestion
topic|Name of the topic in Pulsar|plain text|my_topic|/
```

File: docs/reference-guide/streams/overview.md [hunk 3]
```diff
@@ -40,18 +54,42 @@ transform procedure|Name of the transformation file followed by a function name|
...
+transform procedure|Name of the transformation file followed by a function name|function|my_transformation.my_transform|/
+consumer group|Name of the consumer group in Memgraph|plain text|my_group|mg_consumer
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/streams/overview.md:77]:
```suggestion
```

File: docs/reference-guide/streams/overview.md [hunk 4]
```diff
@@ -40,18 +54,42 @@ transform procedure|Name of the transformation file followed by a function name|
...
-- the processing of the previous batch is completed
-- the previous batch interval ended without receiving any messages
+* the stream is started
+* the processing of the previous batch is completed
+* the previous batch interval ended without receiving any messages
 
 The user who executes the `CREATE` query is going to be the owner of the stream.
 Authentication and authorization are not supported in Memgraph Community, thus
-the owner will always be `Null`, and the privileges are not checked in Memgraph
+the owner will always be `Null` , and the privileges are not checked in Memgraph
```
Comment (antaljanosbenjamin) [inline docs/reference-guide/streams/overview.md:92]: Do we need the extra space before the comma?

File: docs/database-functionalities/streams/pulsar/pulsar-streams.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+CHECK STREAM myStream BATCH_LIMIT 3 TIMEOUT 60000;
+```
+
+As a result, you should see multiple messages (probably 3) in the output:
+
+```plaintext
++---------------------------------------------------------------+--------------------------------------------------
+| query                                                         | parameters                                      |
++---------------------------------------------------------------+--------------------------------------------------
+| "CREATE (n:MESSAGE {payload: $payload, topic: $topic})"       | {payload: "Memgraph <3 Kafka", topic: "topic1"} |
```
Comment (antaljanosbenjamin) [inline docs/database-functionalities/streams/pulsar/pulsar-streams.md:108]:
```suggestion
| "CREATE (n:MESSAGE {payload: $payload, topic: $topic})"       | {payload: "Memgraph <3 Pulsar", topic: "topic1"} |
```
Important :smile:

File: docs/database-functionalities/streams/kafka/kafka-streams.md [hunk 6]
```diff
@@ -46,10 +46,6 @@ Check the created stream:
 SHOW STREAMS;
 ```
 
-:::warning
-If you're running this in Memgraph Lab, an arbitrary error might happen. Please ignore it until we resolve the issue.
-:::
-
 The result should be similar to:
 
 ```plaintext
```
Comment (antaljanosbenjamin) [inline docs/database-functionalities/streams/kafka/kafka-streams.md:51]: Please update the table below.

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]
Please also search for `CREATE STREAM` in the docs, because there more occurences:
 - in the `docs/import-data/kafka` folder
 - in `docs/database-functionalities/streams/kafka/kafka-streams.md` at line 38 (I couldn't comment on it :disappointed:)
 - in docs/reference-guide/security.md at line 153

File: docs/database-functionalities/overview.md [hunk 7]
```diff
@@ -5,53 +5,75 @@ sidebar_label: Database functionalities overview
...
+Articles within this section serve as a cookbook for getting things done as fast
+as possible. These articles tend to provide a step-by-step guide on how to use
+certain Memgraph features or solve a particular problem.
+
+## Streams
+
+Memgraph can connect to existing **Kafka**, **Pulsar** and **Redpanda** streams.
+The streams are using user-defined transformations to produce Cypher queries
+based on the received messages. To learn more about streams and transformations
+take a look at the following pages:
```
Comment (jbajic) [inline docs/database-functionalities/overview.md:17]:
```suggestion
Articles within this section serve as a cookbook for getting things done as fast
as possible. These articles tend to provide a step-by-step guide on using
certain Memgraph features or solving a particular problem.

## Streams

Memgraph can connect to existing **Kafka**, **Pulsar** and **Redpanda** streams.
The streams use user-defined transformations to produce Cypher queries
based on the received messages. To learn more about streams and transformations
take a look at the following pages:
```

File: docs/database-functionalities/streams/pulsar/implement-transformation-module.md [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+properties of each message in a vertex.
+
+:::note
+For detailed technical information on transformation modules, check out the [reference guide](/reference-guide/streams/transformation-modules/overview.md).
+:::
+
+## Using Docker with transformation modules
+
+If you are using Docker to run Memgraph, you will have to create a volume
+and mount it to access the `query_modules` directory. Yes, `query_modules` ,
```
Comment (jbajic) [inline docs/database-functionalities/streams/pulsar/implement-transformation-module.md:19]:
```suggestion
and mount it to access the `query_modules` directory. Yes, `query_modules`,
```

File: docs/database-functionalities/streams/pulsar/implement-transformation-module.md [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+extern "C" int mgp_init_module(mgp_module *module, mgp_memory *memory) {
+  return mgp_module_add_transformation(module, "my_c_transformation",
+                                       my_c_transformation) == 0;
+}
+```
+
+Now let's compile it:
+
+```shell
+clang++ --std=c++17 -Wall -shared -fPIC -I /home/kovi/data/memgraph/include c_transformation.cpp -o c_transformation.so
```
Comment (jbajic) [inline docs/database-functionalities/streams/pulsar/implement-transformation-module.md:252]:
```suggestion
clang++ --std=c++17 -Wall -shared -fPIC -I /path/to/include c_transformation.cpp -o c_transformation.so
```

File: docs/database-functionalities/streams/pulsar/pulsar-streams.md [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+
+## Creating the stream
+
+The very first step is to make sure at least one transformation module is loaded
+into Memgraph. If you are not sure how to define them, check out the
+[transformation module
+guide](/database-functionalities/streams/pulsar/implement-transformation-module.md).
+We will use `transformation.my_pulsar_transformation` from that example, but we
+are going to alias it as `my.pulsar_transform` to make the size of result tables
+slimmer. For the topic name, we are going to use `topic1` .
```
Comment (jbajic) [inline docs/database-functionalities/streams/pulsar/pulsar-streams.md:32]:
```suggestion
slimmer. For the topic name, we are going to use `topic1`.
```

File: docs/database-functionalities/streams/pulsar/pulsar-streams.md [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+Maybe at first, you don't want to run the stream in the background but see the
+actual result of the transformation. This can be handy when implementing a
+transformation. To achieve that, we can use the `CHECK STREAM` query. This query
+will consume the message from the last acknowledged message but won't
+acknowledge the next message. That means you are free to play around with it,
+and there won't be any permanent effects. For a freshly created stream, there is
+probably no acknowledged message, so the `CHECK STREAM` query will wait for new
+messages. By default, the query will wait `30000` milliseconds ( `30` seconds)
+and after that, it will throw a timeout exception. To give us some more time,
+use a larger timeout, e.g.: `60000` milliseconds ( `60` seconds):
```
Comment (jbajic) [inline docs/database-functionalities/streams/pulsar/pulsar-streams.md:79]: Is the behaviour for pulsar the same? Does it consume the message?

Comment (antonio2368) [reply docs/database-functionalities/streams/pulsar/pulsar-streams.md:78]: yes it is, no acknowledgment is done.

**Pull request merged**